### ISSUE TITLE
Master build broken due to failed unit test

### ISSUE THREAD

**Issue by ghentschke** [eclipse/lsp4e#930]
The CI build job fails due to a filed unit test: 
```
array lengths differed, expected.length=3 actual.length=0; arrays first differed at element [0]; expected:<StyleRange {0, 4, fontStyle=normal, foreground=Color {133, 153, 0, 255}}> but was:<end of array>
```
The failed test is 
https://github.com/eclipse/lsp4e/blob/78fffa519b11b5c56bcc612e5133a4cef8adfb95/org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/semanticTokens/SemanticHighlightReconcilerStrategyTest.java#L51

**Comment by ghentschke** [comment]
Has anyone an idea why this test is now failing?

**Comment by mickaelistria** [comment]
@ghentschke Can you reproduce the failure locally? Can you find the offending change that has triggered it?

**Comment by ghentschke** [comment]
Yes, I can reproduce it on my machine. I'll take a look.

**Comment by ghentschke** [comment]
This test failure has been caused by a commit in tm4e. I opened[ this tm4e issue](https://github.com/eclipse/tm4e/issues/721), but I am not sure if its should be fixed in tm4e or in lsp4e.

**Comment by ghentschke** [comment]
It could be fixed in lp4e by #940

**Comment by ghentschke** [comment]
Fixing one issue and two others popping up. This [unit test](https://github.com/eclipse/lsp4e/blob/master/org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/references/FindReferencesTest.java) seems to be flaky. They fail with *Widget is disposed* error.
